themes/origins/assets/component-testimonials.js

```diff
@@ -0,0 +1,65 @@
+class Testimonials extends HTMLElement {
+  static observedAttributes = ["product-id"];
+
+  constructor() {
+    super();
+    this.PRODUCT_ID = this.getAttribute("product-id");
+
+    this.container = this.querySelector("[data-container]");
+    this.skeleton = this.querySelector("[data-skeleton]");
+    this.empty = this.querySelector("[data-empty]");
+  }
+
+  connectedCallback() {
+    this._render();
+  }
+
+  async _render() {
+    try {
+      const response = await youcanjs.product.fetchReviews(this.PRODUCT_ID).data();
+
+      response.length ? this.setupItems(response) : this.isEmpty();
+    } catch (error) {
+      console.error(error);
+
+      this.isEmpty();
+      this.PRODUCT_ID && toast.show(error.message, "error");
```

Comment:
i think this is an unnecessary condition
```suggestion
      toast.show(error.message, "error");
```

Reply:
try adding this section without assigning a `product`

Reply:
We can refrain from sending the request instead (only send request if a product_id exists)